src/components/tx/TxSimulation/useSimulation.ts

```diff
@@ -38,17 +38,25 @@ export const useSimulation = (): UseSimulationReturn => {
   }, [])
 
   const simulateTransaction = useCallback(
-    async (params: Omit<SimulationTxParams, 'gasLimit'>) => {
+    async (params: Omit<SimulationTxParams, 'gasLimit'> & { manualGasLimit?: number }) => {
       if (!web3ReadOnly) return
 
       setSimulationRequestStatus(FETCH_STATUS.LOADING)
       setRequestError(undefined)
 
       try {
-        const { gasLimit } = await web3ReadOnly.getBlock('latest')
+        const manualGasLimit = params.manualGasLimit
+        let gasLimit: number
+        if (manualGasLimit) {
+          gasLimit = manualGasLimit
+        } else {
+          const latestBlock = await web3ReadOnly.getBlock('latest')
```

Comment:
I think it's unlikely that there's no gas limit passed? Do we need this fallback?

Reply:
Yes for batch transactions we don't have gas estimation there. See [ReviewBatchExecute.tsx](https://github.com/safe-global/web-core/blob/main/src/components/tx/modals/BatchExecuteModal/ReviewBatchExecute.tsx#L96)

Reply:
Gotcha 👍